src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,38 @@
 package mate.academy;
 
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private final List<EventListener> eventListeners = new CopyOnWriteArrayList<>();
+    private final ExecutorService executorService = Executors.newCachedThreadPool();
 
+    public void registerListener(EventListener listener) {
+        eventListeners.add(listener);
     }
 
     public void deregisterListener(EventListener listener) {
-
+        eventListeners.remove(listener);
     }
 
     public void notifyEvent(Event event) {
-
+        for (EventListener listener : eventListeners) {
+            executorService.submit(() -> listener.onEvent(event));
+        }
```

Comment:
Submitting tasks to an ExecutorService inside a loop may cause unbounded queue growth if task submission outpaces task execution. Consider using a bounded queue or rate-limiting task submission.